deployment/ccip/changeset/v1_6/cs_add_new_chain_e2e.go

```diff
@@ -0,0 +1,932 @@
+			return fmt.Errorf("failed to validate RMN remote config for new chain: %w", err)
+		}
+	}
+	if err := c.updateChainConfig().Validate(e); err != nil {
+		return fmt.Errorf("failed to validate update chain config: %w", err)
+	}
+
+	return nil
+}
+
+func addCandidatesForNewChainLogic(e deployment.Environment, c AddCandidatesForNewChainConfig) (deployment.ChangesetOutput, error) {
+	newAddresses := deployment.NewMemoryAddressBook()
+	var allProposals []mcmslib.TimelockProposal
+
+	// Save existing contracts
+	err := runAndSaveAddresses(func() (deployment.ChangesetOutput, error) {
+		return commoncs.SaveExistingContractsChangeset(e, c.NewChain.ExistingContracts)
+	}, newAddresses, e.ExistingAddresses)
+	if err != nil {
+		return deployment.ChangesetOutput{}, fmt.Errorf("failed to run SaveExistingContractsChangeset on chain with selector %d: %w", c.NewChain.Selector, err)
+	}
+
+	// Deploy the prerequisite contracts to the new chain
+	err = runAndSaveAddresses(func() (deployment.ChangesetOutput, error) {
+		return changeset.DeployPrerequisitesChangeset(e, c.prerequisiteConfigForNewChain())
+	}, newAddresses, e.ExistingAddresses)
+	if err != nil {
+		return deployment.ChangesetOutput{}, fmt.Errorf("failed to run DeployPrerequisitesChangeset on chain with selector %d: %w", c.NewChain.Selector, err)
+	}
+
+	// Deploy MCMS contracts
+	if c.MCMSDeploymentConfig != nil {
+		err = runAndSaveAddresses(func() (deployment.ChangesetOutput, error) {
+			return commoncs.DeployMCMSWithTimelockV2(e, map[uint64]commontypes.MCMSWithTimelockConfigV2{
+				c.NewChain.Selector: *c.MCMSDeploymentConfig,
+			})
+		}, newAddresses, e.ExistingAddresses)
+		if err != nil {
+			return deployment.ChangesetOutput{}, fmt.Errorf("failed to run DeployMCMSWithTimelockV2 on chain with selector %d: %w", c.NewChain.Selector, err)
+		}
+	}
+
+	// Deploy chain contracts to the new chain
+	err = runAndSaveAddresses(func() (deployment.ChangesetOutput, error) {
+		return DeployChainContractsChangeset(e, c.deploymentConfigForNewChain())
+	}, newAddresses, e.ExistingAddresses)
+	if err != nil {
+		return deployment.ChangesetOutput{}, fmt.Errorf("failed to run DeployChainContractsChangeset on chain with selector %d: %w", c.NewChain.Selector, err)
+	}
+
+	// Set RMN remote config & set RMN on proxy on the new chain (if config provided)
+	if c.NewChain.RMNRemoteConfig != nil {
+		_, err = SetRMNRemoteConfigChangeset(e, c.rmnRemoteConfigForNewChain())
+		if err != nil {
+			return deployment.ChangesetOutput{}, fmt.Errorf("failed to run SetRMNRemoteConfigChangeset on chain with selector %d: %w", c.NewChain.Selector, err)
+		}
+	}
+	// Set the RMN remote on the RMN proxy
+	_, err = SetRMNRemoteOnRMNProxyChangeset(e, SetRMNRemoteOnRMNProxyConfig{
+		ChainSelectors: []uint64{c.NewChain.Selector},
+	})
+	if err != nil {
+		return deployment.ChangesetOutput{}, fmt.Errorf("failed to run SetRMNRemoteOnRMNProxyChangeset on chain with selector %d: %w", c.NewChain.Selector, err)
+	}
```

Comment:
In most cases RMNProxy would be pre-owned. How do we handle that?

Reply:
https://github.com/smartcontractkit/chainlink/pull/16705/commits/8104b00de8d2820f96e1d73a532a8a32c98f51f0

Was thinking about this too, I just added a conditional for this one. If owned by MCMS, pass MCMS config.